### PR TITLE
Fix the behavior that was broken

### PR DESCRIPTION
**Pull request by guipublic**
by the merge of the ssa refactoring into function-return PR, the previous behavior is restored but it should be refactored again when we will support multiple return values.

### CODE REVIEW

**Comment by jfecher** [comment]
What behavior was broken that this fixes?

**Comment by guipublic** [comment]
So the code that I modified in this PR was in the #221 PR before I merged the refactoring. It enable support for functions that are used without assigning them, for instance `constrain f(x) == 3`.
The point is I did all the tests with this code , which is valid for functions that return only one value, so this is both working and required.
Obviously this is not good for multiple return values and this will be fixed. I have the multiple values PR ready, I just need these fix PRs to be merged first.
Long story short; this is a quick fix that will be cleaned with the multiple return PR.
